sdk/terraform/Azure.ResourceManager.Terraform/src/Generated/Extensions/TerraformExtensions.cs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Terraform.Mocking;
+using Azure.ResourceManager.Terraform.Models;
+
+namespace Azure.ResourceManager.Terraform
+{
+    /// <summary> A class to add extension methods to Azure.ResourceManager.Terraform. </summary>
+    public static partial class TerraformExtensions
+    {
+        private static MockableTerraformSubscriptionResource GetMockableTerraformSubscriptionResource(ArmResource resource)
+        {
+            return resource.GetCachedClient(client => new MockableTerraformSubscriptionResource(client, resource.Id));
+        }
+
+        /// <summary>
+        /// Exports the Terraform configuration of the specified resource(s).
+        /// <list type="bullet">
+        /// <item>
+        /// <term>Request Path</term>
+        /// <description>/subscriptions/{subscriptionId}/providers/Microsoft.AzureTerraform/exportTerraform</description>
+        /// </item>
+        /// <item>
+        /// <term>Operation Id</term>
+        /// <description>AzureTerraform_ExportTerraform</description>
+        /// </item>
+        /// <item>
+        /// <term>Default Api Version</term>
+        /// <description>2023-07-01-preview</description>
+        /// </item>
+        /// </list>
+        /// <item>
+        /// <term>Mocking</term>
+        /// <description>To mock this method, please mock <see cref="MockableTerraformSubscriptionResource.ExportTerraformAzureTerraformClient(WaitUntil,BaseExportModel,CancellationToken)"/> instead.</description>
+        /// </item>
+        /// </summary>
+        /// <param name="subscriptionResource"> The <see cref="SubscriptionResource" /> instance the method will execute against. </param>
+        /// <param name="waitUntil"> <see cref="WaitUntil.Completed"/> if the method should wait to return until the long-running operation has completed on the service; <see cref="WaitUntil.Started"/> if it should return after starting the operation. For more information on long-running operations, please see <see href="https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/LongRunningOperations.md"> Azure.Core Long-Running Operation samples</see>. </param>
+        /// <param name="exportParameter"> The export parameter. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="subscriptionResource"/> or <paramref name="exportParameter"/> is null. </exception>
+        public static async Task<ArmOperation<OperationStatus>> ExportTerraformAzureTerraformClientAsync(this SubscriptionResource subscriptionResource, WaitUntil waitUntil, BaseExportModel exportParameter, CancellationToken cancellationToken = default)
```

Comment:
So the spec might not be specifying the correct information for this long-running operation.
From the client's perspective, it makes no sense that this returns `OperationStatus`, instead it should return the actual payload of this operation, `ExportResult`.

Reply:
Also we need a `@clientName` to add on this method in the spec in a `client.tsp` file to rename this operation to `ExportTerraform`.

Reply:
@ArcturusZhang `ExportResult` is a child property of `OperationStatus` (refer to [`OperationStatus.properties`](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/terraform/Microsoft.AzureTerraform.Management/models.tsp#L122)).

I tried making the generated SDK code returns `ArmOperation<ExportResult>` but I get error (expected response payload does not match actual).  ([spec change](https://github.com/Azure/azure-rest-api-specs/compare/main...gerrytan:azure-rest-api-specs:lro-change-final-state-to-exportresult?expand=1) - [SDK diff](https://github.com/gerrytan/azure-sdk-for-net/compare/terraform-export-sdk-generation-2...gerrytan:azure-sdk-for-net:lro-change-final-state-to-exportresult?expand=1) - error on dotnet test [dotnet-test-azure-rm-terraform.log](https://github.com/user-attachments/files/17302695/dotnet-test-azure-rm-terraform.log))

Do you have any suggestion how to improve this further? Or perhaps an example API that is similar to ours?

Reply:
Have improved the method name with `@clientName("ExportTerraform")` on 811553076c1. Spec change is pending ([branch](https://github.com/Azure/azure-rest-api-specs/compare/main...gerrytan:azure-rest-api-specs:more-terraform-spec-improvements-for-net-sdk?expand=1)). I will PR and merge once the generated SDK passes review.